internal/plugin/install.go

```diff
@@ -0,0 +1,154 @@
+package plugin
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"syscall"
+)
+
+type PluginInstall struct {
+	Enabled bool `json:"enabled"`
+
+	// Current active version of the plugin
+	Version string `json:"version"`
+
+	// Map of a plugin version to the extracted directory
+	ExtractedVersions map[string]string `json:"extracted_versions"`
+
+	manifest       *PluginManifest
+	runningVersion *string
+	processManager *ProcessManager
+	rpcListener    *net.Listener
```

Comment:
`net.Listener` is already an interface (which is a pointer) so you don't need to make this a pointer to be able to check for `nil`.